plugins/backstage-plugin-devcontainers-backend/src/processors/DevcontainersProcessor.test.ts

```diff
@@ -0,0 +1,233 @@
+} from '@backstage/backend-common';
+import { NotFoundError } from '@backstage/errors';
+import type { LocationSpec } from '@backstage/plugin-catalog-common';
+import {
+  type Entity,
+  ANNOTATION_SOURCE_LOCATION,
+} from '@backstage/catalog-model';
+import {
+  DEFAULT_TAG_NAME,
+  DevcontainersProcessor,
+  PROCESSOR_NAME_PREFIX,
+} from './DevcontainersProcessor';
+
+const mockUrlRoot = 'https://github.com/example-company/example-repo';
+
+const baseEntity: Entity = {
+  apiVersion: 'backstage.io/v1alpha1',
+  kind: 'Component',
+  metadata: {
+    name: 'metadata',
+    tags: [], // Purposefully left empty
+    annotations: {
+      [ANNOTATION_SOURCE_LOCATION]: `${mockUrlRoot}/tree/main`,
+    },
+  },
+};
+
+const baseLocation: LocationSpec = {
+  type: 'Component',
+  presence: 'required',
+  target: `${mockUrlRoot}/blob/main/catalog-info.yaml`,
+};
+
+type SetupOptions = Readonly<{
+  tagName?: string;
+
+  searchThrowCallback?: (
+    url: string,
+    readOptions: ReadUrlOptions | undefined,
+  ) => never;
+
+  readUrlThrowCallback?: (
+    url: string,
+    readOptions: ReadUrlOptions | undefined,
+  ) => never;
+}>;
+
+function setupProcessor(options?: SetupOptions) {
+  const {
+    searchThrowCallback,
+    readUrlThrowCallback,
+    tagName = DEFAULT_TAG_NAME,
+  } = options ?? {};
+
+  /**
+   * Tried to get this working as more of an integration test that used MSW, but
+   * couldn't figure out how to bring in the right dependencies in time. So this
+   * is more of a unit test for now (which might be all we really need?)
+   *
+   * Likely candidates for making this work are ConfigReader from
+   * @backstage/config or GithubCredentialsProvider from @backstage/integrations
+   *
+   * setupRequestMockHandlers from @backstage/backend-test-utils will be helpful
+   * for hooking up MSW
+   */
+  const mockReader = {
+    readTree: jest.fn(),
+    readUrl: jest.fn(async (url, readOptions): Promise<ReadUrlResponse> => {
+      readUrlThrowCallback?.(url, readOptions);
+
+      return {
+        buffer: jest.fn(),
+        stream: jest.fn(),
+      };
+    }),
+    search: jest.fn(async (url, readOptions): Promise<SearchResponse> => {
+      searchThrowCallback?.(url, readOptions);
+
+      return {
+        etag: readOptions?.etag ?? 'fallback etag',
+        files: [
+          {
+            content: () => Promise.resolve(new Buffer('blah')),
+            url: mockUrlRoot,
+          },
+        ],
+      };
+    }),
+  } as const satisfies UrlReader;
+
+  const processor = new DevcontainersProcessor(mockReader, {
+    tagName,
+    logger: getVoidLogger(),
+  });
+
+  return { mockReader, processor } as const;
+}
+
+describe(`${DevcontainersProcessor.name}`, () => {
+  describe('getProcessorName', () => {
+    it('Should use Coder prefix in the output', () => {
+      const { processor } = setupProcessor();
+      const name = processor.getProcessorName();
+      expect(name).toMatch(new RegExp(`^${PROCESSOR_NAME_PREFIX}`));
```

Comment:
Edit to note that currently we have `backstage-plugin-devcontainers-backend//devcontainers-processor` (double slash), definitely not a bug or anything but might not be what was intended.

Reply:
So, it's a preemptive thing (and maybe I should remove it because we don't need it yet), but it is possible to have multiple instances of the same processor, and have the processor name be partially dynamic

I guess I was trying to add a test case that would be more future-proofed, but that extra slash absolutely wasn't intentional. Will remove